onnxruntime/core/framework/path_lib.h

```diff
@@ -6,58 +6,87 @@
 #include <string>
 #include <sstream>
 #include <assert.h>
+#include <stdexcept>
 #ifdef _WIN32
 #include <Windows.h>
 #include <pathcch.h>
 #else
 #include <libgen.h>
 #include <sys/types.h>
 #include <dirent.h>
+#include <stddef.h>  //ptrdiff_t
 #endif
 #include "core/session/onnxruntime_c_api.h"
 
+using PATH_CHAR_TYPE = ORTCHAR_T;
+
+template <typename T>
+long OrtStrtol(const T* nptr, T** endptr, int base);
+
+/**
+ * Convert a C string to ssize_t(or ptrdiff_t)
+ * @return the converted integer value.
+ */
+template <typename T>
+ptrdiff_t OrtStrtoPtrDiff(const T* nptr, T** endptr, int base);
```

Comment:
>OrtStrtoPtrDiff [](start = 10, length = 15)

Would it be better to put these OrtStr... functions somewhere more generic? They're not really specific to processing a path given they're used for processing 'offset' and 'length' values in tensorprotoutils.

Do we need 'base' in the API or can we just hardcode that to 10?

nit: OrtStrToPtrDiff